src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -384,6 +384,14 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
                     }
                 }
             }
+
+            if (!bindingPoint.HasNewValue && !bindingPoint.IsReadOnly && bindingPoint.Value is not null)
```

Comment:
In the 6.0 code, we always called the setter in `BindProperty` if the value was not null:

https://github.com/dotnet/runtime/blob/5cfe5d815ccb49187ba392b8f1a169b233dc21bc/src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs#L263-L266

Should we instead do something more like that to fix this issue?

Reply:
This will partially fix the issue but will miss other cases. Initially I had the fix in the exact place you pointed at, but I discovered this is not addressing the issue when using ICollection. 

When calling [BindCollection](https://github.com/dotnet/runtime/blob/fd9f52098bba9e88269b2b147a45b8f60e4b8d0d/src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs#L379), inside BindCollection we'll run into same problem not calling the setter [here](https://github.com/dotnet/runtime/blob/fd9f52098bba9e88269b2b147a45b8f60e4b8d0d/src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs#L656).  I preferred fixing the issue in one common place addressing all known cases and ensuring it will apply to other cases we may be missing too.

Reply:
> inside BindCollection we'll run into same problem not calling the setter [here](https://github.com/dotnet/runtime/blob/fd9f52098bba9e88269b2b147a45b8f60e4b8d0d/src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs#L656).

That line runs for each item in the collection. Not the property itself. `BindProperty` still runs for the property, right?

Reply:
Looks you are right. The collection property binding should run through BindProperty. I modified the code according your suggestion.